ROS2/ROS_workspace/cli_app/src/gui.cpp

```diff
@@ -0,0 +1,414 @@
+            show_video.store(true);
+        }
+
+        ImGui::End();
+        
+
+        if (glfwJoystickPresent(GLFW_JOYSTICK_1)) {
+            int axesCount, buttonsCount;
+            const float* axes = glfwGetJoystickAxes(GLFW_JOYSTICK_1, &axesCount);
+            const unsigned char* buttons = glfwGetJoystickButtons(GLFW_JOYSTICK_1, &buttonsCount);
+
+            if (axesCount >= 5) { // Ensure there are enough axes
+                input.surge = static_cast<int8_t>(axes[1] * -127); // Left thumbstick vertical axis
+                input.sway = static_cast<int8_t>(axes[0] * 127);  // Left thumbstick horizontal axis
+                input.pitch = static_cast<int8_t>(axes[4] * 127); // Right thumbstick horizontal axis
+                input.yaw = static_cast<int8_t>(axes[3] * 127); // Right thumbstick horizontal axis
+            }
+
+            for (int i = 0; i < buttonsCount; i++) {
+                bool button_state = (buttons[i] == GLFW_PRESS);
+                switch (i) {
+                    case 0: break;
+                    case 1: break;
+                    case 2: input.heave_up = button_state; break;
+                    case 3: input.heave_down = button_state; break;
+                    case 4: input.pitch_up = button_state; break;
+                    case 5: input.pitch_down = button_state; break;
+                    case 6: input.open_claw = button_state; break;
+                    case 7: input.close_claw = button_state; break;
+                    case 8: break;
+                    case 9: break;
+                    case 10: break;
+                    case 11: break;
+                    case 12: break;
+                    case 13: input.heave_up = button_state; break;
+                    case 14: break;
+                    case 15: input.heave_down = button_state; break;
+                    case 16: break;
+                    default: break;
+                }
+            }
+
+            ImGui::Begin("Controller Input");
+            ImGui::Text("Axes:");
+            ImGui::Text("Surge: %d", input.surge);
+            ImGui::Text("Sway: %d", input.sway);
+            ImGui::Text("Heave: %d", input.heave);
+            ImGui::Text("Pitch: %d", input.pitch);
+            ImGui::Text("Yaw: %d", input.yaw);
+
+            ImGui::Text("Buttons:");
+            ImGui::Text("Open Claw: %s", input.open_claw ? "Pressed" : "Released");
+            ImGui::Text("Close Claw: %s", input.close_claw ? "Pressed" : "Released");
+            ImGui::Text("Heave Up: %s", input.heave_up ? "Pressed" : "Released");
+            ImGui::Text("Heave Down: %s", input.heave_down ? "Pressed" : "Released");
+            ImGui::Text("Pitch Up: %s", input.pitch_up ? "Pressed" : "Released");
+            ImGui::Text("Pitch Down: %s", input.pitch_down ? "Pressed" : "Released");
+            ImGui::Text("Brighten LED: %s", input.brighten_led ? "Pressed" : "Released");
+            ImGui::End();
```

Comment:
There seems to be some duplicated functionality in this file with `controller.cpp` (with respect to the controller input publisher and associated code)

I understand that `controller.cpp` functionality is meant to be called when you provide the option "launch_controller" instead of "launch_gui". However, can we make it so that "launch_gui" itself references `controller.cpp` instead of duplicating it's functionality?